book/crud.rst

```diff
@@ -0,0 +1,477 @@
+
+        /**
+         * @ORM\Column(type="integer")
+         */
+        private $seats;
+
+        /**
+         * @ORM\Column(type="date")
+         */
+        private $boughtAt;
+
+        /**
+         * @ORM\Column(type="date")
+         */
+        private $leasedUntil;
+
+        public function getId()
+        {
+            return $this->id;
+        }
+
+        public function getModel()
+        {
+            return $this->model;
+        }
+
+        public function setModel($model)
+        {
+            $this->model = $model;
+        }
+
+        public function getSeats()
+        {
+            return $this->seats;
+        }
+
+        public function setSeats($seats)
+        {
+            $this->seats = $seats;
+        }
+
+        public function getBoughtAt()
+        {
+            return $this->boughtAt;
+        }
+
+        public function setBoughtAt($boughtAt)
+        {
+            $this->boughtAt = $boughtAt;
+        }
+
+        public function getLeasedUntil()
+        {
+            return $this->leasedUntil;
+        }
+
+        public function setLeasedUntil($leasedUntil)
+        {
+            $this->leasedUntil = $leasedUntil;
+        }
+    }
+
+.. _book-crud-controller-class:
+
+The main Controller Class
+-------------------------
+
+The controller actions to view, create and modify entities that will be explained in the following
+sections will live in the ``VehicleController`` class:
+
+.. code-block:: php
+    :linenos:
+
+    // src/InventoryBundle/Controller/VehicleController.php
+    namespace InventoryBundle\Controller;
+
+    use Sensio\FrameworkExtraBundle\Configuration\Route;
+    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+    /**
+     * @Route("/vehicle")
+     */
+    class VehicleController extends Controller
+    {
+    }
+
+Unless in common Symfony applications you only have to add the routing information for this
+controller class in a ``routing.yml`` file that is located in the bundle's ``Resources/config/oro``
+directory:
+
+.. code-block:: yaml
+    :linenos:
+
+    # src/InventoryBundle/Resources/config/oro/routing.yml
+    inventory_bundle:
+        resource: "@InventoryBundle/Controller"
+        type: annotation
+        prefix: /inventory
+
+When the routing configuration is located in the ``Resources/config/oro`` directory, it will be
+discovered automatically when the cache is warmed up. So you do not need to add anything to the
+global application routing configuration.
+
+.. _book-crud-grid:
+
+The Datagrid
+------------
+
+Relying on the features provided by the Oro Platform, the controller for listing the stored objects
+becomes really slim. It basically just needs to return the type of object being displayed in the
+grid:
+
+.. code-block:: php
+    :linenos:
+
+    // src/InventoryBundle/Controller/VehicleController.php
+    namespace InventoryBundle\Controller;
+
+    use Sensio\FrameworkExtraBundle\Configuration\Template;
```

Comment:
- Should be Sensio**\Bundle**\FrameworkExtraBundle\Configuration\Template
- Also the import for Acl annotation is missing:
```php
use Oro\Bundle\SecurityBundle\Annotation\Acl;
```